### ISSUE TITLE
Returning 404 error from JSON route generates 500 error

### ISSUE THREAD

**Issue by sentience** [arc-repos/arc-functions#6]
[According to the docs](https://arc.codes/guides/http#errors), calling `res` with an `Error` object lets you return a 404 error if you set the `code` property of the error object. I'm attempting to do this as follows in a `@json` route handler:

```js
let arc = require("@architect/functions");

function route(req, res) {
    const error = Error("Record not found");
    error.code = 404;
    res(error);
}

exports.handler = arc.json.get(route);
```

With `npx sandbox`, requesting this route crashes the server with an error:

```
/Users/kyank/code/project/node_modules/aws-sdk/lib/request.js:31
            throw err;
            ^

Error: Record not found
    at Object.route (/Users/kyank/code/project/src/json/get-record-000recordId/index.js:22:19)
    at _iter (/Users/kyank/code/project/src/json/get-surveys-000surveyId/node_modules/@architect/functions/src/http/_request.js:73:12)
    at _read (/Users/kyank/code/project/src/json/get-surveys-000surveyId/node_modules/@architect/functions/src/http/_request.js:74:9)
    at _find (/Users/kyank/code/project/src/json/get-surveys-000surveyId/node_modules/@architect/functions/src/http/session/read.js:37:5)
    at Response._get (/Users/kyank/code/project/src/json/get-surveys-000surveyId/node_modules/@architect/functions/src/http/session/_find.js:17:9)
    at Request.<anonymous> (/Users/kyank/code/project/node_modules/aws-sdk/lib/request.js:364:18)
    at Request.callListeners (/Users/kyank/code/project/node_modules/aws-sdk/lib/sequential_executor.js:109:20)
    at Request.emit (/Users/kyank/code/project/node_modules/aws-sdk/lib/sequential_executor.js:81:10)
    at Request.emit (/Users/kyank/code/project/node_modules/aws-sdk/lib/request.js:683:14)
    at Request.transition (/Users/kyank/code/project/node_modules/aws-sdk/lib/request.js:22:10)
```

With the project deployed to AWS, requesting this route returns a 500 error, and logs this in CloudWatch:

```
{
    "errorMessage": "{\"statusCode\":404,\"json\":{\"message\":\"Record not found\",\"code\":404,\"time\":{},\"name\":\"Error\",\"stack\":\"Error: Record not found\\n    at Object.route (/var/task/index.js:22:19)\\n    at _iter (/var/task/node_modules/@architect/functions/src/http/_request.js:73:12)\\n    at _read (/var/task/node_modules/@architect/functions/src/http/_request.js:74:9)\\n    at _find (/var/task/node_modules/@architect/functions/src/http/session/read.js:37:5)\\n    at Response._get (/var/task/node_modules/@architect/functions/src/http/session/_find.js:17:9)\\n    at Request.<anonymous> (/var/runtime/node_modules/aws-sdk/lib/request.js:364:18)\\n    at Request.callListeners (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:105:20)\\n    at Request.emit (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:77:10)\\n    at Request.emit (/var/runtime/node_modules/aws-sdk/lib/request.js:683:14)\\n    at Request.transition (/var/runtime/node_modules/aws-sdk/lib/request.js:22:10)\"}}"
}
```

**Comment by sentience** [comment]
Worked around this by avoiding the use of `Error` to respond with an error, and instead return a normal response with a `status` field:

```js
```

**Comment by brianleroux** [comment]
👋 Hi @sentience thanks for reporting! We just shipped arc v4.0 with a bunch of improvements including a fix that restores the ability to do this (albeit in a slightly different way). Want to apologize that you caught us in between releases here and we recently deprecated and moved a bunch of stuff around as a result. I hope you like the changes!

All the API Gateway functions are now under `@http` in the `.arc` file:

```javascript
@app
myfunapp

@http
get /
get /page
get /page/:subpage
post /login
get /css
get /js
```
This would generate a bunch of (local only!) code: `npx create local`
To test these new routes out: `npx sandbox`

The default method signature is now dependency free and async/await friendly. 

```javascript
exports.handler = async function http(req) {
  return {
    type: 'text/html',
    body: '<b>hello world!</b>'
  }
}
```

But we kept a very similar method signature around for ppl porting from older arc apps or express apps. You can opt into it as before with `@architect/functions`.

```javascript
let arc = require('@architect/functions')
function route(req, res) {
  res({html: '<i>hello world!!</i>'})
}
exports.handler = arc.http(route)
```
> Note: the signature is now `arc.http`

I *just* added back the ability to pass an error to `res` too

```javascript
let arc = require('@architect/functions')
function route(req, res) {
  let err = Error('whoopsy')
  err.code = 503
  res(err)
}
exports.handler = arc.http(route)
```
Super sorry for the thrash! We're a young project and learning fast what is working and is not. And thanks again for reporting / super appreciate any feedbacks you have!

**Comment by sentience** [comment]
Thanks for following up! Very happy with the new release, and looking forward to giving it a try this week. Totally understand the realities of young projects like this, and appreciate your responsiveness to feedback. :)
